Sources/TuistLoader/Mappers/ProjectNameAndOrganizationMapper.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import ProjectDescription
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// It's a mapper that changes the project name and organization based on the configuration
+/// in the Config.swift
+public class ProjectNameAndOrganizationMapper: ProjectMapping {
+    private let config: TuistCore.Config
+
+    public init(config: TuistCore.Config) {
+        self.config = config
+    }
+
+    // MARK: - ProjectMapping
+
+    public func map(project: TuistCore.Project) throws -> (TuistCore.Project, [SideEffectDescriptor]) {
+        var project = project
+
+        // Xcode project file name
+        var xcodeProjPath: AbsolutePath = project.xcodeProjPath
+        if let xcodeFileName = xcodeFileNameOverride(for: project) {
+            xcodeProjPath = project.xcodeProjPath.parentDirectory.appending(component: "\(xcodeFileName).xcodeproj")
+        }
+        project.xcodeProjPath = xcodeProjPath
```

Comment:
minor:

This could be simplified slightly

there isn't a need for the additional `xcodeProjPath` local variable 

```swift
if let xcodeFileName = xcodeFileNameOverride(for: project) {
      project.xcodeProjPath = project.xcodeProjPath.parentDirectory.appending(component: "\(xcodeFileName).xcodeproj")
}
```